src/runtime/audience-action-flow.ts

```diff
@@ -321,6 +327,68 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   **/
+  private async configureAnswerPpsData(
+    request: SurveyDataTransferRequest
+  ): Promise<void> {
+    const iabAudienceKey = StorageKeys.PPS_TAXONOMIES;
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()!
+      .flatMap((question) => question.getSurveyAnswersList())
+      .map((answer) => answer?.getPpsValue())
+      .filter((ppsValue) => ppsValue !== null);
+
+    if (ppsConfigParams.length === 0 || !ppsConfigParams) {
+      log(`[swg.js] No PPS values found in survey answer choices.`);
+    }
+
+    const existingIabTaxonomy = await this.storage_.get(
+      iabAudienceKey,
+      /* useLocalStorage= */ true
+    );
+    let ppsConfigParamsValues;
+    if (!existingIabTaxonomy) {
```

Comment:
Can we define a variable for the existing pps values (an array of string, which can be empty in case there's nothing) and append the current values to the array to construct a new array to override stored value without having to deal with if/else statements?

Reply:
makes sense. will do that